framework/src/problems/FEProblemBase.C

```diff
@@ -4097,7 +4097,7 @@ FEProblemBase::reinitMaterialsFace(const SubdomainID blk_id,
                                    const bool swap_stateful,
                                    const std::deque<MaterialBase *> * const reinit_mats)
 {
-  if (hasActiveMaterialProperties(tid))
+  if (hasActiveMaterialProperties(tid) && needSubdomainMaterialOnSide(blk_id, tid))
```

Comment:
comment in ComputeMaterialsObjectThread explains why I cant fold the hasActiveMatProp call into needSubdomainMaterialXYZ class. The needSubdomainMaterialXYZ are called at a time we dont know who's needed (=active)

Reply:
ended up being necessary because of the stateful vs regular computation.

the name  of the routine could likely be improved